### PR TITLE
clean up paths before uploading

### PR DESCRIPTION
**Pull request by C0rby**
Just a small bugfix to clean up the file paths.

- [ ] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

**Comment by jesmrec** [comment]
@abelgardep @JuancaG05 need to check if this fixes applies correctly to the `new_arch/synchronization` branch (from where new release will be done)

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java [hunk 1]
```diff
@@ -960,6 +968,8 @@ public void afterTextChanged(Editable editable) {
                             MAX_FILENAME_LENGTH);
                 } else if (fileName.length() == 0) {
                     error = getString(R.string.uploader_upload_text_dialog_filename_error_empty);
+                } else if (fileName.contains("/")) {
+                   error = getString(R.string.filename_forbidden_characters);
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java:972]: We introduced a [FileNameValidator on the new architecture](https://github.com/owncloud/android/blob/new_arch%2Fsynchronization/owncloudDomain/src/main/java/com/owncloud/android/domain/validator/FileNameValidator.kt) branch to handle every validation the same way. It is not available on master yet, so we will handle this when we merge/rebase with new architecture

**Review by abelgardep — APPROVED** [review]

**Comment by abelgardep** [comment]
> @abelgardep @JuancaG05 need to check if this fix applies correctly to the `new_arch/synchronization` branch (from where new release will be done)

Shouldn't be a problem, that Activity haven't got major changes in the new architecture. We can merge this into master or new_arch with no issues.

**Comment by jesmrec** [comment]
Tests perform over the following upload use cases:

over `master:

- Single manual upload root/non-root
- Multiple manual upload root/non-root
- Take photo root
- Take photo non-root
- Share from Photos
- Share text with oC
- Automatic uploads (photo/video)

i will recreate those tests over `new_arch/synchronization`, but some conflict happened when pulling the branch.

**Pull request merged**